.github/workflows/qa-tests.yaml

```diff
@@ -16,6 +16,10 @@ jobs:
   CI:
     runs-on: macos-13
 
+    env:
+      # Avoid concurrency issues
+      BROWSERSTACK_CUSTOM_ID: ${{ (github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch') && format('pr-{0}', github.sha) || 'iOSApp' }}
```

Comment:
> the github.sha is not the SHA of the commit that the GH workflow is running against

It is, though it is not the latest commit on the branch, if that is what you were expecting. `github.sha`, which is the default value for the `checkout` action and therefore what the GH workflow runs against, is a [merge commit with the base branch](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) rather than the branch alone. We would have to [override that manually](https://github.com/actions/checkout?tab=readme-ov-file#Checkout-pull-request-HEAD-commit-instead-of-merge-commit) if we wanted to, which I don't think we do.

So, I think it is suitable for us, yes.